src/fstests/runner.py

```diff
@@ -21,20 +21,48 @@ def __init__(self, configs, args):
         self._db_config = configs['db_configs'][args.db_config]
         api_token = os.getenv('API_TOKEN')
         self._db = kernelci.db.get_db(self._db_config, api_token)
+        self._plan_configs = configs['test_plans']
+        self._device_configs = configs['device_types']
+        self._output = args.output
+        if not os.path.exists(self._output):
+            os.makedirs(self._output)
+        runtime_config = configs['labs'][args.lab_config]
+        self._runtime = kernelci.lab.get_api(runtime_config)
 
-    def check_kvm_xfstest_env(self):
+    def _schedule_job(self, node, plan_config, device_config, tmp):
+        revision = node['revision']
+        params = {
+            'db_config_yaml': self._db_config.to_yaml(),
+            'name': plan_config.name,
+            'node_id': node['_id'],
+            'revision': revision,
+            'runtime': self._runtime.config.lab_type,
+            'workspace': tmp,
+        }
+        params.update(plan_config.params)
+        params.update(device_config.params)
+        templates = ['config/runtime', '/etc/kernelci/runtime']
+        job = self._runtime.generate(
+            params, device_config, plan_config, templates_path=templates
+        )
+        output_file = self._runtime.save_file(job, tmp, params)
+        job_result = self._runtime.submit(output_file)
+        return job_result
+
+    def _run_single_job(self, tarball_node, plan, device):
         try:
-            result = subprocess.run("kvm-xfstests --help", check=True, shell=True)
-            if result.returncode == 0:
-                print(f"KVM-xftests found.")
-                return True
-            else:
-                return False
-        except Exception as e:
-            print(f"Raised exception: {e}")
-            return False
+            tmp = tempfile.TemporaryDirectory(dir=self._output)
+            job = self._schedule_job(tarball_node, plan, device, tmp.name)
+            if self._runtime.config.lab_type == 'shell':
```

Comment:
This and other things can be simplified because it will only be running in `shell`.  Having a separate service from the regular `src/runner.py` means we can have the code tailored to what is really needed for fstests and keep it shorter.